components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.cc

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_page_metrics_observer.h"
+
+#include "brave/components/brave_perf_predictor/browser/perf_predictor_tab_helper.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_contents.h"
+
+namespace brave_perf_predictor {
+
+PerfPredictorPageMetricsObserver::PerfPredictorPageMetricsObserver() = default;
+
+PerfPredictorPageMetricsObserver::
+    ~PerfPredictorPageMetricsObserver() = default;
+
+page_load_metrics::PageLoadMetricsObserver::ObservePolicy
+PerfPredictorPageMetricsObserver::OnCommit(
+    content::NavigationHandle* navigation_handle,
+    ukm::SourceId source_id) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  // Skip if off the record
+  if (navigation_handle->GetWebContents()->
+      GetBrowserContext()->IsOffTheRecord()) {
+    return STOP_OBSERVING;
+  }
+
+  navigation_id_ = navigation_handle->GetNavigationId();
+  // We'll be forwarding all performance metrics to the observer
+  observer_ = PerfPredictorTabHelper::FromWebContents(
+        navigation_handle->GetWebContents());
+  if (!observer_) {
+    VLOG(2) << navigation_id_ << " could not get PerfPredictorTabHelper";
+    return STOP_OBSERVING;
+  }
+  return CONTINUE_OBSERVING;
+}
+
+page_load_metrics::PageLoadMetricsObserver::ObservePolicy
+PerfPredictorPageMetricsObserver::ShouldObserveMimeType(
+    const std::string& mime_type) const {
+  // Observe all MIME types. We still only use actual data usage, so strange
+  // cases (e.g., data:// URLs) will still record the right amount of data
+  // usage.
+  return CONTINUE_OBSERVING;
+}
+
+void PerfPredictorPageMetricsObserver::OnFirstContentfulPaintInPage(
+    const page_load_metrics::mojom::PageLoadTiming& timing) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  if (observer_) {
+    observer_->OnPageLoadTimingUpdated(timing);
+  } else if (VLOG_IS_ON(2)) {
+    VLOG(2) << "PerfPredictorTabHelper not ready for timing updates";
+  }
+  VLOG(3) << navigation_id_ << " paint timing "
+    << "NavigationToFirstContentfulPaint "
+    << timing.paint_timing->first_contentful_paint.value();
+}
+
+void PerfPredictorPageMetricsObserver::
+  OnFirstMeaningfulPaintInMainFrameDocument(
+    const page_load_metrics::mojom::PageLoadTiming& timing) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  if (observer_) {
+    observer_->OnPageLoadTimingUpdated(timing);
+  } else if (VLOG_IS_ON(2)) {
```

Comment:
you don't need VLOG_IS_ON here, the `VLOG` macro takes care of that for you and the macros are even smart enough to avoid running anything inline in the output if VLOG is disabled for that level